### PR TITLE
Allow custom username prefix while creating ephemeral Splunk user

### PR DESCRIPTION
**Pull request by abhide**
Tickets: TE-34

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,7 +93,7 @@ Create temporary admin account:
     password           439e831b-e395-9999-2cd7-856381db3394
     roles              [admin]
     url                https://localhost:8089
-    username           vault_local-admin_okta-mweber_70c6c140-238d-e12b-3289-8e38f8c4d9f5_1553712516020311000
+    username           vault_70c6c140-238d-e12b-3289-8e38f8c4d9f5
 
 This creates a new user account `vault_local-admin_okta-mweber_70c6...`
```
Comment (michaelw) [inline README.md]: nit: update

File: path_creds_create.go [hunk 2]
```diff
@@ -62,7 +60,7 @@ func (b *backend) credsReadHandler(ctx context.Context, req *logical.Request, d
 	if err != nil {
 		return nil, err
 	}
-	username := fmt.Sprintf("vault_%s_%s_%s_%d", name, req.DisplayName, userUUID, time.Now().UnixNano())
+	username := fmt.Sprintf("%s_%s", role.UserPrefix, userUUID)
```
Comment (michaelw) [inline path_creds_create.go]: userUUID might not be unique, e.g., did you test same user requesting two emphemeral creds with overlapping lifetime?

Comment (michaelw) [reply path_creds_create.go]: also, should we keep requestor name in there for tracking? (at least unless explicitly overridden)

File: README.md (hunk 1 above)
Comment (abhide) [reply README.md]: Done

File: path_creds_create.go (hunk 2 above)
Comment (abhide) [reply path_creds_create.go]: A new UUID is generated for userUUID for each new user. Check 3 lines above

Comment (abhide) [reply path_creds_create.go]: Added req.DisplayName to username

**Review by michaelw — APPROVED** [review]
we might want to document that `defaultUserPrefix` is a special value now (or not make it special)

**Pull request merged**